src/main/java/lotto/domain/WinLotto.java

```diff
@@ -3,31 +3,33 @@
 import java.util.List;
 
 public class WinLotto {
-    private final Numbers numbers;
-    private final Integer bonusNumber;
+    private final LottoNumbers lottoNumbers;
+    private final LottoNo bonusNumber;
 
-    public WinLotto(Numbers numbers) {
-        this(numbers, null);
+    public WinLotto(LottoNumbers lottoNumbers) {
+        this(lottoNumbers, null);
     }
 
-    public WinLotto(Numbers numbers, Integer bonusNumber) {
-        this.numbers = numbers;
+    public WinLotto(LottoNumbers lottoNumbers, LottoNo bonusNumber) {
+        this.lottoNumbers = lottoNumbers;
         this.bonusNumber = bonusNumber;
     }
 
-    public WinLotto(List<Integer> numbers, Integer bonusNumber) {
-        this(new Numbers(numbers), bonusNumber);
+    public WinLotto(List<LottoNo> numbers, LottoNo bonusNumber) {
+        this(new LottoNumbers(numbers), bonusNumber);
     }
 
-    public Numbers numbers() {
-        return numbers;
+    public LottoNumbers numbers() {
+        return lottoNumbers;
     }
 
-    public Integer findNumber(int i) {
-        return numbers.find(i);
+    public LottoNo bonusNumber() {
+        return bonusNumber;
     }
 
-    public Integer bonusNumber() {
-        return bonusNumber;
+    public Rank checkRank(Lotto lotto) {
+        int matchCount = this.lottoNumbers.countContains(lotto.numbers());
+        boolean matchBounus = lotto.numbers().contains(this.bonusNumber);
```

Comment:
디미터 법칙을 위반하고 있습니다.
lottoNumbers와 lotto가 직접 처리할 수 있을 것 같아요
```suggestion
        int matchCount = this.lottoNumbers.countContains(lotto);
        boolean matchBounus = lotto.hasNumber(this.bonusNumber);
```